### PR TITLE
Remote console elaboration

### PR DESCRIPTION
**Pull request by aefimov3**
## What does this PR do?

Remote console elaboration: add --console command line key, send command from user typing, improve disconnection handling

## How was this PR tested?

Local test: iPhone + Windows PC

### CODE REVIEW

File: Tools/RemoteConsole/remote_console.py [hunk 1]
```diff
@@ -5,54 +5,91 @@
...
 def main():
-    global remoteConsole
+    global remote_console
+    global client_message_logger
+    global messages_to_console
 
     parser = argparse.ArgumentParser(description="Provides common dev functionality", add_help=False)
     parser.add_argument('--addr', default='127.0.0.1', dest='address', help="Address for Remote Console to connect")
     parser.add_argument('--port', default='4600', dest='port', help="Port for Remote Console to connect")
+    parser.add_argument('--console', action='store_true', dest='console', help="All the messages to console instead of remote_console.log file")
```
Comment (JasonCarbonated) [inline Tools/RemoteConsole/remote_console.py]: The help message here is a little vague. This sends commands to the remote console when using --console correct? It doesn't just cause messages that would normally go to the remote_console.log file to be displayed in the console, right? Also sends commands yes?

**Review by JasonCarbonated — CHANGES_REQUESTED** [review]
Does using --console stop remote_console.log from logging? It looks like it would. Should we have it also generate a remote_console.log?

File: Tools/RemoteConsole/remote_console.py (hunk 1 above)
Comment (aefimov3) [reply Tools/RemoteConsole/remote_console.py]: --console is to print client's log messages to console. Junho wanted this.
Sending commands is independent from the --console command line key, it works any case.

**Review by Twolewis — APPROVED** [review]

**Comment by aefimov3** [comment]
@JasonCarbonated 
"Does using --console stop remote_console.log from logging? It looks like it would. Should we have it also generate a remote_console.log?"
I'm not sure, I believe Junho wanted it to be print to console. It is possible to redirect console output to a file if needed using tee.
Anyway this is pretty easy to change, having a file too does not harm, I believe.

**Pull request merged**